### ISSUE TITLE
[BUG] Unable to set custom CSS for Python decorators

### ISSUE THREAD

**Issue by tsirakisn** [NaturalDocs/NaturalDocs#41]
According to https://www.naturaldocs.org/reference/styles/, the .SHMetadata class can be used to set syntax highlighting for python's @decorator tags. Setting this class has no effect on the decorator output, however. I verified that I am able to modify other SH classes (e.g. strings, numbers, etc) properly, so my methodology appears correct.

Is this an error in the documentation or should this be possible?

---------------------------------------------------
My CSS file (ModifyDecorators.css):
```
.SHMetadata {
    color: #FFFFFF;
    font-style: italic;
}
```

**Comment by NaturalDocs** [comment]
Sorry, class prototypes are a completely different animal than regular prototypes.  The documentation doesn't make that clear at all so I'll update it later.  For now, the class you want to apply that style to is .CPPrePrototypeLine

**Comment by tsirakisn** [comment]
@NaturalDocs thank you for your response. The CPPrePrototypeLine class doesn't seem to work either, can you confirm my process? I have the line `Style: ModifyDecorators` in my Project.txt and I have the following in ModifiyDecorators.css (which is in the same folder as Project.txt):

```
.CPPrePrototypeLine {
    font-style: italic;
}
```

I see no change in the decorators being shown. As an extra bit of info, my end goal is to be able to hide decorators from the documentation entirely, so if you have a different means of doing that, please let me know.

**Comment by NaturalDocs** [comment]
Try adding !important afterwards, since the default styles may be more specific than just .CPPrePrototypeLine and therefore may be winning out.

```
.CPPrePrototypeLine {
    font-style: italic !important;
}
```

To hide them you can try display: none !important.

**Comment by ntsirakis** [comment]
@NaturalDocs sorry for the late response, the above didn't seem to work either. Is there a current effort to improve the formatting of python functions/classes that use decorators? That's really the main concern. Here's an example of the generating format I'm seeing:
```
@decorator @decorator_with_args(arg1 = val,
                                                          arg2 = val) @other_decorator def fn(self)
```

The alternative to the above is to sandwich the ND docs between the decorators and the function:

```
@decorator1
@decorator2
# Method: fn
# Some description.
# 
# Parameter: arg1
def fn(self, arg1):
```

The above causes the decorators to not appear in the documentation at all, which is great, but the code itself is not very readable.

**Comment by NaturalDocs** [comment]
This is fixed in Natural Docs 2.1 Development Release 2.  You can download it here:

https://www.naturaldocs.org/download#Development

It covers both your test cases without needing to sandwich the comment between the decorators and the function definition.

As for hiding the decorators in the output, I verified that this will work:
````
.NDClassPrototype .CPPrePrototypeLine,
.NDPrototype .PParameterSection,
.NDPrototype .PPlainSection {
   display: none;
   }
.NDPrototype .PParameterSection:last-child,
.NDPrototype .PPlainSection:last-child {
   display: block;
   }
````
If you want to just change the style, you would do this instead:
````
.NDPrototype .SHMetadata,
.NDClassPrototype .SHMetadata {
   font-style: italic;
   }
````
The first example is done at the block level instead of at .SHMetadata because using the second approach with display: none will leave some space behind if the decorators have parameters like in your first example.

**Comment by ntsirakis** [comment]
@NaturalDocs tried the above and it worked like a charm! Thanks a lot for the fix.
